pysumma/preprocessing/create_local_attribute.py

```diff
@@ -0,0 +1,237 @@
+            fdr.shape[1], fdr.shape[0])
+        hruid = pd.DataFrame(hruid_array)
+        hruid = hruid.T
+        downHRUindex = pd.DataFrame(np.zeros((hruid.shape[0], hruid.shape[1])))
+        downHRUindex.astype('int64').head()
+        for index_r in range(fdr.shape[0]):
+            for index_c in range(fdr.shape[1]):
+                if fdr.loc[index_r, index_c] == 1 and index_r-1 >= 0 and index_c+1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r-1, index_c+1]
+                elif fdr.loc[index_r, index_c] == 2 and index_r-1 >= 0 and index_c >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r-1, index_c]
+                elif fdr.loc[index_r, index_c] == 3 and index_r-1 >= 0 and index_c-1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r-1, index_c-1]
+                elif fdr.loc[index_r, index_c] == 4 and index_r >= 0 and index_c-1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r, index_c-1]
+                elif fdr.loc[index_r, index_c] == 5 and index_r+1 >= 0 and index_c-1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r+1, index_c-1]
+                elif fdr.loc[index_r, index_c] == 6 and index_r+1 >= 0 and index_c >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r+1, index_c]
+                elif fdr.loc[index_r, index_c] == 7 and index_r+1 >= 0 and index_c+1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r+1, index_c+1]
+                elif fdr.loc[index_r, index_c] == 8 and index_r >= 0 and index_c+1 >= 0:
+                    downHRUindex.loc[index_r,
+                                     index_c] = hruid.loc[index_r, index_c+1]
+                elif fdr.loc[index_r, index_c] < 0 and index_r >= 0 and index_c+1 >= 0:
+                    downHRUindex.loc[index_r, index_c] = -9999
+                else:
+                    downHRUindex.loc[index_r, index_c] = 0
+        downHRUindex = downHRUindex.astype('int64')
+        downHRUindex_one_col = pd.concat([pd.DataFrame(downHRUindex.loc[:, [i]].values, columns=[
+                                         downhruindex_col_name]) for i in range(len(downHRUindex.columns))], ignore_index=True)
+        df_format[downhruindex_col_name] = downHRUindex_one_col.values
+        return df_format
+
+    def lulc_index(self, lulc_df, option='USGS'):
+        lulc = pd.DataFrame(
+            np.zeros((lulc_df.shape[0], lulc_df.shape[1])), columns=['vegTypeIndex'])
+        lulc = lulc.astype('int64')
+        if option == "USGS" and "USGS-RUC":
```

Comment:
I think the logic here is not what you want. The `and "USGS-RUC"` part doesn't actually do anything. I think you mean `if option == "USGS" or option == "USGS-RUC"`, which could also be written `option in ["USGS", "USGS-RUC"]`.